### DISCUSSION TITLE
How run qt applications under X in wayfire?

### DISCUSSION THREAD

**Discussion by LinArcX** [WayfireWM/wayfire#1546] [Q&A]
Hi.

I want to run flameshot in wayfire, but it gives me errors:

```
↳ flameshot
flameshot: error: Unable to detect desktop environment (GNOME? KDE? Sway? ...)
flameshot: error: Hint: try setting the XDG_CURRENT_DESKTOP environment variable.
flameshot: error: Unable to capture screen
flameshot: error: Unable to capture screen
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
```

I think if I run it under X, it'll be ok. But I'm not sure. What's you idea?

**Comment by Ozwald1** [comment]
Just give him env XDG_CURRENT_DESKTOP=Sway. Do local override copy of Flameshot'ts desktop file and modify EXEC lines.. 

cp /usr/share/applications/org.flameshot.Flameshot.desktop .local/share/applications/

and all Exec lines like: Exec=env XDG_CURRENT_DESKTOP=Sway flameshot

And report this issue to Flameshot's devs too, of course that should work without trick on Wayfire too ;)

**Reply by Ozwald1** [reply]
Yet, Flameshot is native Wayland app since it's not possible X apps would take snipped shreenshots from Wayland.

**Reply by LinArcX** [reply]
I did this:
`XDG_CURRENT_DESKTOP=Sway`

But by clicking on flameshot, (in system tray, since I'm using waybar) it won't take any screenshot nor throw any error.

**Reply by Ozwald1** [reply]
Hmm, what is your version? I can confirm 12.1.0 works here, but earlier didn't..

**Reply by Ozwald1** [reply]
Anyway, this is Flameshot's issue

**Reply by LinArcX** [reply]
```
↳ flameshot --version
Flameshot v12.1.0 (-)
Compiled with Qt 5.15.4
```

**Comment by ammen99** [comment]
You can make a wrapper script around Wayfire and set `XDG_CURRENT_DESKTOP=Sway` there.
